common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -822,7 +827,9 @@ public MergeStatuses finalizeShuffleMerge(FinalizeShuffleMerge msg) {
                 msg.appAttemptId, msg.shuffleId, msg.shuffleMergeId, partition.reduceId,
                 ioe.getMessage());
           } finally {
-            partition.closeAllFilesAndDeleteIfNeeded(false);
+            AppShufflePartitionInfo.closeAllFilesAndDeleteIfNeeded(false, partition.dataFile,
```

Comment:
```
CLEANER.register(this, new ResourceCleaner(dataFile, dataChannel, indexFile,
        metaFile, appAttemptShuffleMergeId, reduceId));
```
Maybe we can assign it to `Cleaner.Cleanable cleanable`, and then we can change this to call `partitionInfo.cleanable.clean();` there.

And then, IIRC,  if `partitionInfo.cleanable.clean();` is called manually,` ResouceCleaner.run` will not be executed repeatedly.